tools/generators/migrate-v8-pkg/index.spec.ts

```diff
@@ -1,20 +1,269 @@
 import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing';
-import { Tree, readProjectConfiguration } from '@nrwl/devkit';
+import {
+  Tree,
+  readProjectConfiguration,
+  readWorkspaceConfiguration,
+  WorkspaceConfiguration,
+  serializeJson,
+  stripIndents,
+  addProjectConfiguration,
+  ProjectConfiguration,
+  logger,
+} from '@nrwl/devkit';
+import type { Linter } from 'eslint';
 
+import type { TsConfig } from '../../types';
 import generator from './index';
 import { MigrateV8PkgGeneratorSchema } from './schema';
 
+interface AssertedSchema extends MigrateV8PkgGeneratorSchema {
+  name: string;
+}
+
 describe('migrate-v8-pkg generator', () => {
-  let appTree: Tree;
-  const options: MigrateV8PkgGeneratorSchema = { name: 'test' };
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  const noop = () => {};
+
+  let tree: Tree;
+  const options = { name: '@proj/eight' } as const;
 
   beforeEach(() => {
-    appTree = createTreeWithEmptyWorkspace();
+    jest.restoreAllMocks();
+
+    jest.spyOn(console, 'log').mockImplementation(noop);
+    jest.spyOn(console, 'info').mockImplementation(noop);
+    jest.spyOn(console, 'warn').mockImplementation(noop);
+
+    tree = createTreeWithEmptyWorkspace();
+    tree = setupDummyPackage(tree, options);
+    tree = setupDummyPackage(tree, {
+      name: '@proj/react',
+      eslintConfig: stripIndents`
+     module.exports = {
+       extends: ['@proj/react--legacy'],
+       root: true
+     }
+    `,
+    });
+    tree = setupDummyPackage(tree, { name: '@proj/merge-styles' });
   });
 
   it('should run successfully', async () => {
-    await generator(appTree, options);
-    const config = readProjectConfiguration(appTree, 'test');
+    await generator(tree, options);
+    const config = readProjectConfiguration(tree, options.name);
     expect(config).toBeDefined();
   });
+
+  describe(`--stats`, () => {
+    beforeEach(() => {
+      setupDummyPackage(tree, {
```

Comment:
standard setup mostly copied from `migrate-converged-pkg` with small modifications for v8. we can unify and reuse later if appropriate